api/provider/aws/job_runner.go

```diff
@@ -1,25 +1,256 @@
+	case job.CreateServiceJob:
+		return r.createService(j.JobID, j.Request)
+	case job.CreateTaskJob:
+		return r.createTask(j.JobID, j.Request)
+	case job.DeleteDeployJob:
+		return r.deleteDeploy(j.JobID, j.Request)
+	case job.DeleteEnvironmentJob:
+		return r.deleteEnvironment(j.JobID, j.Request)
+	case job.DeleteLoadBalancerJob:
+		return r.deleteLoadBalancer(j.JobID, j.Request)
+	case job.DeleteServiceJob:
+		return r.deleteService(j.JobID, j.Request)
+	case job.DeleteTaskJob:
+		return r.deleteTask(j.JobID, j.Request)
+	case job.UpdateEnvironmentJob:
+		return r.updateEnvironment(j.JobID, j.Request)
+	case job.UpdateLoadBalancerJob:
+		return r.updateLoadBalancer(j.JobID, j.Request)
+	case job.UpdateServiceJob:
+		return r.updateService(j.JobID, j.Request)
+	default:
+		return fmt.Errorf("Unrecognized JobType '%s'", j.Type)
+	}
+}
+
+func (r *JobRunner) createDeploy(jobID, request string) error {
+	var req models.CreateDeployRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	deploy, err := r.deployProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, deploy.DeployID)
+}
+
+func (r *JobRunner) createEnvironment(jobID, request string) error {
+	var req models.CreateEnvironmentRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	environment, err := r.environmentProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, environment.EnvironmentID)
+}
+
+func (r *JobRunner) createLoadBalancer(jobID, request string) error {
+	var req models.CreateLoadBalancerRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	loadBalancer, err := r.loadBalancerProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, loadBalancer.LoadBalancerID)
+}
+
+func (r *JobRunner) createService(jobID, request string) error {
+	var req models.CreateServiceRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	service, err := r.serviceProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, service.ServiceID)
+}
+
+func (r *JobRunner) createTask(jobID, request string) error {
+	var req models.CreateTaskRequest
+	if err := json.Unmarshal([]byte(request), &req); err != nil {
+		return errors.New(errors.InvalidRequest, err)
+	}
+
+	task, err := r.taskProvider.Create(req)
+	if err != nil {
+		return err
+	}
+
+	return r.jobStore.SetJobResult(jobID, task.TaskID)
+}
+
+func (r *JobRunner) deleteDeploy(jobID, deployID string) error {
+	return r.deployProvider.Delete(deployID)
+}
+
+func (r *JobRunner) deleteEnvironment(jobID, environmentID string) error {
+	loadBalancers, err := r.loadBalancerProvider.List()
```

Comment:
On an initial look this doesn't seem right. The implementation details about how an environment should delete itself should be encapsulated at the environment level. But this isn't strictly an environment delete so I can see the case for why the method contains logic for deleting environment and all related entities.